New sensor cert_expiry

fabfurnari opened the pull request:
## Simple sensor to check HTTPS certificate expiry (in days)

This simple sensor fetches a certificate using the `ssl` module anc check the `notAfter` directive in HTTPS certificate. Reports then the expiry in days.

## Example entry for `configuration.yaml` (if applicable):
```yaml

sensor:
  - platform: cert_expiry
    server_name: home-assistant.io
    server_port: 443

```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code does not interact with devices:
  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

fabfurnari committed:
New sensor cert_expiry

fabfurnari committed:
New sensor cert_expiry

fabfurnari committed:
New sensor cert_expiry

homeassistant commented:
Hi @fabfurnari, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#7207).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob commented:
Nice! This is ok to merge when you fix the linting issue and add it to `.coveragerc` (you removed that step from your PR description…)

amelchio committed:
Support xy_color with LIFX lights (#7208)

biacz committed:
Update ios.py (#7160)

* Update ios.py

as discussed. the part: 
       if battery_state == ios.ATTR_BATTERY_STATE_FULL:
            returning_icon_level = DEFAULT_ICON_LEVEL
kinda screws up the charging icon.

i might just miss a logical solution for that though.
let me know what you think. it might not be beautiful but i think its an overall improve over the current "double battery" solution

* Update ios.py

chound fix and full_battery_charge fix

* Update ios.py

removed new line

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* merged request from robbie

* Update ios.py

* Update ios.py

* Update ios.py

sdague committed:
Fix arwn platform to update hass state when events are received (#7202)

The arwn platform was refactored to be asyncio friendly, however in
doing so one thing was missed which was explicitly telling hass when
something interesting has happened. This led to the very interesting
to debug issue that the state cards were all out of date, even though
the graphs were not.

hmn committed:
Issue 6749 updated pylgtv to 0.1.6 to fix thread leak in asyncio loop (#7199)

* updated pylgtv module to fix problems with timeouts

* - update pylgtv to 0.1.6
- handle new TimeoutError exception from pylgtv

* used full name for exception handling of concurrent.futures._base.TimeoutError

* the exception handling should now follow the rules

* float typecasting should not be necessary

* use asyncio for TimeoutError it’s an alias for concurrent.futures.TimeoutError

balloob committed:
tradfri: Improve color temp support detection (#7211)

fabfurnari committed:
Added cert_expiry to coveragerc

fabfurnari committed:
Added cert_expiry to coveragerc

fabfurnari committed:
cert_expiry: fixed some pylint complaints

fabfurnari committed:
cert_expiry: fixed some pylint complaints

fabfurnari committed:
fixed conflict

fabfurnari commented:
@balloob thanks, I'll work on documentation as soon as I have some other spare time!

fabaff commented on homeassistant/components/sensor/cert_expiry.py:
```diff
@@ -0,0 +1,98 @@
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SERVER_NAME = 'server_name'
+CONF_SERVER_PORT = 'server_port'
```
We already have `host` and `port` for those two configuration variables which are defined in `const.py`.

fabaff commented on homeassistant/components/sensor/cert_expiry.py:
```diff
@@ -0,0 +1,98 @@
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SERVER_NAME): cv.string,
+    vol.Optional(CONF_SERVER_PORT, default=443): cv.port,
```
Please consider to add:

```python
vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
```
THis way the users are allowed to name the sensor after their preferences.

fabaff commented on homeassistant/components/sensor/cert_expiry.py:
```diff
@@ -0,0 +1,98 @@
+
+
+class SSLCertificate(Entity):
+    """Implements certificate expiry sensor."""
```
This docstring is not valid according PEP257. I guess that will become an issue with CI.

fabaff commented on homeassistant/components/sensor/cert_expiry.py:
```diff
@@ -0,0 +1,98 @@
+            ctx = ssl.create_default_context()
+            sock = ctx.wrap_socket(socket.socket(),
+                                   server_hostname=self.server_name)
+            sock.connect((self.server_name, self.server_port))
```
I think that you missed to add `sock.settimeout()` as `TIMEOUT` is present.

MartinHjelmare committed:
Fix tradfri lights (#7212)

* Remove leftover use of slugify

* The IKEA manufacturer key is now exactly as found in device info.

* Fix bitwise addition of supported features

fabaff committed:
Bump version to 0.44.0.dev0 (#7217)

jotunacorn committed:
Pushbullet notification sensor (#7182)

* Added the pushbullet sensor component

* Updated requirements_all.txt and .coveragerc with the new sensor

* Updated acording to houndci-bots comments

* Some more changes

* Final change by the hound (?)

* Fixes from balloobs review and from houndci-bot

This changes the sensors information to only contain one attribute
as information, and the rest as device_state_attributes.

* Added leading space to comments

* Added docstrings, removed API_KEY from log, changed imports

* The hound is at it again

* Fix remaining issues

* Fix pylint issue

turbokongen committed:
Zwave cover workaround for graber shades. (#7204)

* wierd pylint complaint

* Workaround for Graber csz1 shades

* logging

* Try direct

* Try direct

* Use workaround

* Review changes and tests

* test

* reset test

* Use Bright and Dim also as open and close is

fabfurnari committed:
Code cleanup and fixes from review

Using correct timeout settings
Logging parameters according to pylint
Using homeassistant const defaults
Using the CONF_NAME default
Minor fixes

fabfurnari committed:
fixed conficts

fabfurnari commented:
thanks to @fabaff for the code review, I've fixed most of the parts.
I checked with pep257 and the docstrings seems ok.

fabaff commented:
This docstring will lead to D401.

Can you rebase your branch please? The PR should only include the related changes. Thanks.

enzymes commented:
Hey Fabrizio,

Great new sensor! Is it possible to monitor more than one certificate in this version?

  - platform: cert_expiry
    host: server1.com

  - platform: cert_expiry
    host: server2.com


Thanks!

balloob commented:
Don't use pull requests for support. Use chat or forums instead.
